### PR TITLE
feat: service logs workflow v2

### PR DESCRIPTION
**Pull request by yesnault**

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19401.0 ✘  
* Build
  * Build and Package All ✘  
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19403.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19405.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19406.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

File: engine/worker/internal/types.go [hunk 1]
```diff
@@ -119,10 +119,14 @@ func (wk *CurrentWorker) GetContext() context.Context {
 	return wk.currentJobV2.context
 }
 
-func (wk *CurrentWorker) SetContext(c context.Context) {
+func (wk *CurrentWorker) SetContextForTestJobV1(c context.Context) {
```
Comment (sguiheux) [inline engine/worker/internal/types.go:122]: add t *testing.T to avoid using it outside test

File: engine/worker/internal/types.go [hunk 2]
```diff
@@ -119,10 +119,14 @@ func (wk *CurrentWorker) GetContext() context.Context {
...
 	wk.currentJob.context = c
 }
 
+func (wk *CurrentWorker) SetContextForTestJobV2(c context.Context) {
```
Comment (sguiheux) [inline engine/worker/internal/types.go:126]: Same

File: engine/api/services/http.go [hunk 3]
```diff
@@ -89,7 +89,7 @@ func doStreamRequest(ctx context.Context, srvs []sdk.Service, method, path strin
 			srv := &srvs[i]
 			reader, headers, code, err := doStreamFromURL(ctx, srv, method, path, bytes.NewReader(b), mods...)
 			if err != nil {
-				lastErr = err
+				lastErr = sdk.WrapError(err, "service:%v", srv.Name)
```
Comment (fsamin) [inline engine/api/services/http.go:92]: you should add metadata as string in the error message, but on the context. The lastErr is print in this function

File: sdk/v2_workflow.go [hunk 4]
```diff
@@ -187,6 +188,19 @@ func (v *V2JobIntegrations) IsEmpty() bool {
 	return v.Artifacts == "" && v.Deployment == ""
 }
 
+type V2JobService struct {
+	Image     string                `json:"image" jsonschema_description:"Docker Image"`
```
Comment (sguiheux) [inline sdk/v2_workflow.go:192]: missing order

File: sdk/v2_workflow.go [hunk 5]
```diff
@@ -187,6 +188,19 @@ func (v *V2JobIntegrations) IsEmpty() bool {
...
+	Env       map[string]string     `json:"env,omitempty" jsonschema_description:"Environment variables"`
```
Comment (sguiheux) [inline sdk/v2_workflow.go:193]: missing order

File: sdk/v2_workflow.go [hunk 6]
```diff
@@ -187,6 +188,19 @@ func (v *V2JobIntegrations) IsEmpty() bool {
...
+	Readiness V2JobServiceReadiness `json:"readiness,omitempty" jsonschema_description:"Service readiness"`
```
Comment (sguiheux) [inline sdk/v2_workflow.go:194]: missing order

File: ui/src/app/views/projectv2/run/run-job-logs.component.ts [hunk 7]
```diff
@@ -195,28 +224,30 @@ export class RunJobLogsComponent {
...
-        if (step?.lines?.length > 0 || step.open) {
-            step.clickOpen();
+    async clickOpen(logBlock: LogBlock) {
+        if (logBlock?.lines?.length > 0 || logBlock.open) {
+            logBlock.clickOpen();
             return;
         }
 
-        step.loading = true;
+        console.log('clickOpen: logBlock', logBlock);
```
Comment (sguiheux) [inline ui/src/app/views/projectv2/run/run-job-logs.component.ts:233]: remove console.log

File: ui/src/app/views/projectv2/run/run-job-logs.component.ts [hunk 8]
```diff
@@ -195,28 +224,30 @@ export class RunJobLogsComponent {
...
+        console.log('clickOpen: logBlock.link', logBlock.link);
```
Comment (sguiheux) [inline ui/src/app/views/projectv2/run/run-job-logs.component.ts:234]: same

File: engine/api/v2_hatchery.go [hunk 9]
```diff
@@ -156,17 +156,24 @@ func (api *API) deleteHatcheryHandler() ([]service.RbacChecker, service.Handler)
...
-				return err
-			}
-
 			tx, err := api.mustDB().Begin()
 			if err != nil {
 				return sdk.WithStack(err)
 			}
 			defer tx.Rollback() // nolint
 
+			hatcheryPermission, err := rbac.LoadRBACByHatcheryID(ctx, api.mustDB(), hatch.ID)
```
Comment (fsamin) [inline engine/api/v2_hatchery.go:165]:
```suggestion
			hatcheryPermission, err := rbac.LoadRBACByHatcheryID(ctx, tx, hatch.ID)
```

**Review by sguiheux — CHANGES_REQUESTED** [review]

File: engine/api/v2_workflow_run_job_routines.go [hunk 10]
```diff
@@ -183,7 +183,7 @@ func (api *API) stopDeadJob(ctx context.Context, store cache.Store, db *gorp.DbM
 	info := sdk.V2WorkflowRunJobInfo{
 		Level:            sdk.WorkflowRunInfoLevelError,
 		WorkflowRunJobID: runJob.ID,
-		Message:          fmt.Sprintf("worker %s don't respond anymore.", runJob.WorkerName),
+		Message:          fmt.Sprintf("worker %s does not respond anymore.", runJob.WorkerName),
```
Comment (fsamin) [inline engine/api/v2_workflow_run_job_routines.go:186]:
```suggestion
		Message:          fmt.Sprintf("worker %q doesn't respond anymore.", runJob.WorkerName),
```

File: engine/worker/internal/types.go (hunk 1 above)
Comment (fsamin) [inline engine/worker/internal/types.go:122]: ????

**Review by fsamin — CHANGES_REQUESTED** [review]
Can you add an integration test ?

File: engine/api/v2_hatchery.go (hunk 9 above)
[the remark this answers is not in the document]
Comment (yesnault) [reply engine/api/v2_hatchery.go:165]: done

File: engine/api/services/http.go (hunk 3 above)
Comment (yesnault) [reply engine/api/services/http.go:92]: log not really useful in fact. (and can't use field in the context, it'a a range of services, and we print err after the loop)

File: engine/api/v2_hatchery.go (hunk 9 above)
Comment (yesnault) [reply engine/api/v2_hatchery.go:165]: done

File: sdk/v2_workflow.go (hunk 4 above)
Comment (yesnault) [reply sdk/v2_workflow.go:192]: done

File: ui/src/app/views/projectv2/run/run-job-logs.component.ts (hunk 8 above)
Comment (yesnault) [reply ui/src/app/views/projectv2/run/run-job-logs.component.ts:234]: done

File: ui/src/app/views/projectv2/run/run-job-logs.component.ts (hunk 7 above)
Comment (yesnault) [reply ui/src/app/views/projectv2/run/run-job-logs.component.ts:233]: done

File: engine/worker/internal/types.go (hunk 1 above)
Comment (yesnault) [reply engine/worker/internal/types.go:122]: SetContextForTestJobV1 dead code, removed

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19413.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19416.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by yesnault** [comment]
> Can you add an integration test ?

yes, incoming

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19471.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19474.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✘  
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19476.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✘  
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19477.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-cds#19555.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19555.0 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

**Review by sguiheux — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-cds#19560.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19560.0 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

**Comment by yesnault** [comment]
> Can you add an integration test ?

done

**Comment by ovh-cds** [comment]
CDS Report build-cds#19573.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19573.0 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✘

**Review by fsamin — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-cds#19582.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19582.0 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✘

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19582.1 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✘

**Comment by ovh-cds** [comment]
CDS Report build-cds#19586.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-cds#19586.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19586.0 ■   
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ■   
  * Standalone Tests v2 ■

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19586.0 ■   
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ■   
  * Standalone Tests v2 ■

**Comment by ovh-cds** [comment]
CDS Report build-cds#19587.0 ✔ 
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#19587.0 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

**Pull request merged**